File: backend/apiswitch/schemas/agents.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class AgentConfigCreate(BaseModel):
    agent_type: str
    config_path: str | None = None
    last_backup_path: str | None = None
    enabled: bool = True
    notes: str | None = None
    settings: dict[str, Any] = Field(default_factory=dict)


class AgentConfigUpdate(BaseModel):
    agent_type: str | None = None
    config_path: str | None = None
    last_backup_path: str | None = None
    enabled: bool | None = None
    notes: str | None = None
    settings: dict[str, Any] | None = None


class AgentConfigRead(BaseModel):
    id: int
    agent_type: str
    config_path: str | None
    last_backup_path: str | None
    enabled: bool
    notes: str | None
    settings: dict[str, Any]
    config_exists: bool
    backup_configured: bool
    created_at: datetime
    updated_at: datetime


class AgentConfigCheckResult(BaseModel):
    ok: bool
    message: str
    config_exists: bool


class ClaudeCodeProfileWriteRequest(BaseModel):
    profile_name: str = "apiswitch"
    base_url: str = "http://127.0.0.1:8080"
    model: str = "code-best"
    effort_level: str | None = None
    max_output_tokens: int | None = Field(default=None, ge=1)
    auto_compact_window: int | None = Field(default=None, ge=1)
    dry_run: bool = False


class ClaudeCodeProfileWriteResult(BaseModel):
    ok: bool
    profile_name: str
    config_dir: str
    settings_path: str
    backup_path: str | None
    written: bool
    settings: dict[str, Any]
    powershell_command: str
    posix_command: str
    message: str

File: frontend/src/api/agents.ts
import { deleteJson, getJson, patchJson, postJson } from './client'

export interface AgentConfig {
  id: number
  agent_type: string
  config_path: string | null
  last_backup_path: string | null
  enabled: boolean
  notes: string | null
  settings: Record<string, unknown>
  config_exists: boolean
  backup_configured: boolean
  created_at: string
  updated_at: string
}

export interface AgentConfigCreate {
  agent_type: string
  config_path?: string | null
  last_backup_path?: string | null
  enabled: boolean
  notes?: string | null
  settings?: Record<string, unknown>
}

export interface AgentConfigUpdate {
  agent_type?: string
  config_path?: string | null
  last_backup_path?: string | null
  enabled?: boolean
  notes?: string | null
  settings?: Record<string, unknown>
}

export interface AgentConfigCheckResult {
  ok: boolean
  message: string
  config_exists: boolean
}

export interface ClaudeCodeProfileWriteRequest {
  profile_name: string
  base_url: string
  model: string
  effort_level?: string | null
  max_output_tokens?: number | null
  auto_compact_window?: number | null
  dry_run: boolean
}

export interface ClaudeCodeProfileWriteResult {
  ok: boolean
  profile_name: string
  config_dir: string
  settings_path: string
  backup_path: string | null
  written: boolean
  settings: Record<string, unknown>
  powershell_command: string
  posix_command: string
  message: string
}

export function fetchAgents() {
  return getJson<AgentConfig[]>('/api/admin/agents')
}

export function createAgent(payload: AgentConfigCreate) {
  return postJson<AgentConfig>('/api/admin/agents', payload)
}

export function updateAgent(agentId: number, payload: AgentConfigUpdate) {
  return patchJson<AgentConfig>(`/api/admin/agents/${agentId}`, payload)
}

export function deleteAgent(agentId: number) {
  return deleteJson<{ deleted: boolean }>(`/api/admin/agents/${agentId}`)
}

export function checkAgent(agentId: number) {
  return postJson<AgentConfigCheckResult>(`/api/admin/agents/${agentId}/check`, {})
}

export function writeClaudeCodeProfile(payload: ClaudeCodeProfileWriteRequest) {
  return postJson<ClaudeCodeProfileWriteResult>('/api/admin/agents/claude-code/write', payload)
}
